pandas/core/arrays/boolean.py

```diff
@@ -296,29 +296,47 @@ def __getitem__(self, item):
             return self._data[item]
         return type(self)(self._data[item], self._mask[item])
 
-    def _coerce_to_ndarray(self, dtype=None, na_value: "Scalar" = libmissing.NA):
+    def to_numpy(self, dtype=None, copy=False, na_value: "Scalar" = libmissing.NA):
         """
-        Coerce to an ndarray of object dtype or bool dtype (if force_bool=True).
+        Convert to a numpy array.
+
+        By default converts to a numpy object array. Specify the `dtype` and
+        `na_value` keywords to customize the conversion.
 
         Parameters
         ----------
         dtype : dtype, default object
-            The numpy dtype to convert to
+            The numpy dtype to convert to.
+        copy : bool, default False
+            Whether to ensure that the returned value is a not a view on
+            the array. Note that ``copy=False`` does not *ensure* that
+            ``to_numpy()`` is no-copy. Rather, ``copy=True`` ensure that
+            a copy is made, even if not strictly necessary.
```

Comment:
Yes, I *think* it is only possible with no NAs and bool dtype. I first thought int8 would also be possible, but numpy doesn't seem to do such conversion without copy.

> And thinking forward, can a pyarrow array with no NAs be converted to an ndarray without any copies?

For boolean not (since it is bits, not bytes), but in general (eg for IntegerArray without nulls) yes